lib/Doctrine/ORM/Mapping/Driver/AttributeDriver.php

```diff
@@ -30,15 +31,24 @@ class AttributeDriver extends CompatibilityAnnotationDriver
 {
     use ColocatedMappingDriver;
 
-    /** @var array<string,int> */
-    // @phpcs:ignore
+    /** @var array<class-string<MappingAttribute>, int> */
+    protected $entityAttributeClasses = [
```

Comment:
Let me know if you think this should be a private property, I don't know why `$entityAnnotationClasses` is exposed or why the class isn't final.

Reply:
`$entityAnnotationClasses` is an extension point of the `AnnotationDriver` from Persistence 2. However, we already override all functionality that uses the property.

Theoretically, you could invent your own attributes and extend `AttributeDriver` to make use of them. Either way, a protected property is a bad extension point. The property is used in the `isTransient()` method only.

My suggestion: Move the array into a private constant, deprecate the property and tell people to override `isTransient()` instead of overriding or changing the property. For the time being, our implementation of `isTransient()` has to use the property, but we can switch it to the new constant on 3.0.x.